### PR TITLE
feat: added image with blur to /leaderboard and /people routes

### LINKED ISSUE #391: Empty profile picture when network is throttled
## Problem
When network is throttled and sometimes in normal mode in /leaderboard, there is a good amount of delay when the profile picture is loaded, sometimes as bad as 5 seconds. This is because the picture is lazy loaded to make the load time of the website faster but at the cost of user experience. Eager loading will just make the load time of site greater.

## Solution
Use a base64url generator like plaiceholder to generate a blur url so that when the picture is being loaded there is a blured picture that has size of less than 1 kB acts as a placeholder for the picture.

### PR DESCRIPTION
**Pull request by Re-Dei**
## Issue Solved:
closes #391 

## Changes:
- added dependencies sharp and plaiceholder
- added new component ImageWithBlur to /people and /leaderboard
- added new method getBase64Url in lib/plaiceholder.ts

## Testing:
tested in development and production in wsl2

### CODE REVIEW

**Review by rithviknishad — CHANGES_REQUESTED** [review]
@Re-Dei 

1. No additional packages are required to achieve blur version of the image... NextJS's Image component itself supports this... Refer: https://nextjs.org/docs/app/api-reference/components/image#placeholder

Lets use this...

3. Getting Internal Server Error upon previewing this deployment...
image

image

**Comment by Re-Dei** [comment]
@rithviknishad yes the nextjs image component does support blur itself but only for images inside the public folder(static images). For dynamic images, such as the github profile pictures, the documentation recommends using plaiceholder to generate the base64url for the blurDataUrl. If i am wrong, please let me know. As for the error, I am currently searching for a solution. Some github issues recommend redeploying without cache to remove the old sharp modules. This error did not exists while running a production build in my local machine.

**Comment by rithviknishad** [comment]
Yes you are right. 

I've trigerred a re-deploy without cache.

**Comment by Re-Dei** [comment]
@rithviknishad did it deploy successfully? Are any changes required?

**Comment by rithviknishad** [comment]
Nope :/ still same error...

**Comment by Re-Dei** [comment]
@rithviknishad the new commit should have fixed the issue. I have tested by deploying in vercel. 
Screenshot (49)

**Review by rithviknishad — CHANGES_REQUESTED** [review]
LGTM, can we use this also for images in `/people` page and all other places too? Preferrably re-use this component in those places.

**Comment by Re-Dei** [comment]
As long as the page is not generate using ISR or the page is static, this can be implemented.

**Comment by Re-Dei** [comment]
As long as the page is not generate using ISR or the page is static, this can be implemented.

File: components/ImageWithBlur.tsx [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import Image, { ImageProps } from "next/image";
+import { getBase64Url } from "@/lib/plaiceholder";
+
+export default async function ImageWithBlur({
+  imageUrl,
```
Comment (rithviknishad) [inline components/ImageWithBlur.tsx:5]: We could just reuse the `src` prop itself right?

Comment (Re-Dei) [reply components/ImageWithBlur.tsx:5]: No, because src could also be the type of StaticImport

Comment (Re-Dei) [reply components/ImageWithBlur.tsx:5]: I think i can have different return for different type of src

Comment (Re-Dei) [reply components/ImageWithBlur.tsx:5]: I merge this branch with main and I am getting this error
⨯ The requested resource isn't a valid image for /logo.webp received text/html; charset=utf-8
ImageError: The requested resource isn't a valid image.
    at imageOptimizer (/workspaces/leaderboard/node_modules/.pnpm/next@14.1.3_react-dom@18.2.0_react@18.2.0/node_modules/next/dist/server/image-optimizer.js:635:19)
    at DevServer.imageOptimizer (/workspaces/leaderboard/node_modules/.pnpm/next@14.1.3_react-dom@18.2.0_react@18.2.0/node_modules/next/dist/server/next-server.js:605:20)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at runNextTicks (node:internal/process/task_queues:64:3)
    at listOnTimeout (node:internal/timers:540:9)
    at process.processTimers (node:internal/timers:514:7)
    at async cacheEntry.imageResponseCache.get.incrementalCache (/workspaces/leaderboard/node_modules/.pnpm/next@14.1.3_react-dom@18.2.0_react@18.2.0/node_modules/next/dist/server/next-server.js:179:65)
    at async /workspaces/leaderboard/node_modules/.pnpm/next@14.1.3_react-dom@18.2.0_react@18.2.0/node_modules/next/dist/server/response-cache/index.js:90:36
    at async /workspaces/leaderboard/node_modules/.pnpm/next@14.1.3_react-dom@18.2.0_react@18.2.0/node_modules/next/dist/lib/batcher.js:41:32 {
  statusCode: 400
}

Comment (Re-Dei) [reply components/ImageWithBlur.tsx]: I tried to merge but this error is stopping me from doing a pnpm build

**Comment by rithviknishad** [comment]
@Re-Dei could you resolve the merge conflicts?

**Comment by Re-Dei** [comment]
I tried to merge this branch with main but i am getting this error
⨯ The requested resource isn't a valid image for /logo.webp received text/html; charset=utf-8
ImageError: The requested resource isn't a valid image.
at imageOptimizer (/workspaces/leaderboard/node_modules/.pnpm/next@14.1.3_react-dom@18.2.0_react@18.2.0/node_modules/next/dist/server/image-optimizer.js:635:19)
at DevServer.imageOptimizer (/workspaces/leaderboard/node_modules/.pnpm/next@14.1.3_react-dom@18.2.0_react@18.2.0/node_modules/next/dist/server/next-server.js:605:20)
at processTicksAndRejections (node:internal/process/task_queues:95:5)
at runNextTicks (node:internal/process/task_queues:64:3)
at listOnTimeout (node:internal/timers:540:9)
at process.processTimers (node:internal/timers:514:7)
at async cacheEntry.imageResponseCache.get.incrementalCache (/workspaces/leaderboard/node_modules/.pnpm/next@14.1.3_react-dom@18.2.0_react@18.2.0/node_modules/next/dist/server/next-server.js:179:65)
at async /workspaces/leaderboard/node_modules/.pnpm/next@14.1.3_react-dom@18.2.0_react@18.2.0/node_modules/next/dist/server/response-cache/index.js:90:36
at async /workspaces/leaderboard/node_modules/.pnpm/next@14.1.3_react-dom@18.2.0_react@18.2.0/node_modules/next/dist/lib/batcher.js:41:32 {
statusCode: 400
}

**Comment by Re-Dei** [comment]
Also in the new feature added to people the image is being fetched twice once for page and once for hover card. Since hover card is only required when hovered, dynamic loading can be used to solve this issue. I am busy for 2 weeks so i can only work on this after 2 weeks. If anyone will be willing to do this, this issue is up for grabs.

**Comment by rithviknishad** [comment]
The way we load org data and related files has changed. Run `pnpm load-data` to get the logo's and other related files placed in the correct place

**Comment by Re-Dei** [comment]
Ok i will try when i am free

**Review by rithviknishad — APPROVED** [review]
LGTM! Nice work!

**Review by rithviknishad — CHANGES_REQUESTED** [review]
Tested the `/people` page btw, doesn't work... https://leaderboard-61azmn2pb-ohcnetwork.vercel.app/people

image

**Comment by Re-Dei** [comment]
the previous error was caused due to default maxDuration being 15s in vercel pro tier. you can check in server logs. I have set the maxDuration to 300s but cannot check if it works because I don't have vercel pro tier. @rithviknishad please check.

**Comment by rithviknishad** [comment]
https://leaderboard-git-fork-re-dei-plaiceholder-ohcnetwork.vercel.app/people

The deploy preview page for `/people` seems to be taking way longer time to load (more than a minute or so), and the blur barely lasts for a second.

**Comment by Re-Dei** [comment]
Should I remove this feature for /people route? I think it works well in /leaderboard route.

**Comment by rithviknishad** [comment]
Even for the leaderboard page, the time has increased by two times. Production deployment loaded in ~5 seconds, but preview deployment leaderboard page loaded in ~10 seconds.

**Comment by Re-Dei** [comment]
Ok. I will close this PR.

**Pull request closed without merging**